sys/net/gnrc/transport_layer/tcp/gnrc_tcp_option.c

```diff
@@ -61,14 +63,15 @@ int _option_parse(gnrc_tcp_tcb_t *tcb, tcp_hdr_t *hdr)
                 break;
 
             default:
-                DEBUG("gnrc_tcp_option.c : _option_parse() : Unknown option found.\
-                      KIND=%"PRIu8", LENGTH=%"PRIu8"\n", option->kind, option->length);
-                return -1;
+                if (TCP_OPTION_LENGTH_MIN <= opt_left) {
+                    DEBUG("gnrc_tcp_option.c : _option_parse() : Unsupported option found.\
+                          KIND=%"PRIu8", LENGTH=%"PRIu8"\n", option->kind, option->length);
+                }
         }
 
         if (opt_left < TCP_OPTION_LENGTH_MIN || option->length > opt_left) {
-            DEBUG("gnrc_tcp_option.c : _option_parse() : invalid option length\n");
-            return 0;
+            DEBUG("gnrc_tcp_option.c : _option_parse() : Invalid option. Drop Packet.\n");
+            return -1;
```

Comment:
Is it okay to merge the two different return values here?

Reply:
Well the 'return 0' is actually part of the buggy behavoir. If the condition comes true, the given options violates the basic option structure either by being to small or exceeding the remaining buffer. In both cases the packet is malformed.

Reply:
The return 0 is part of the buggy behavoir. The condition comes true in cases where an option is either to short or exceeds the given buffer. In both cases the Packet is malformed and has to be dropped in response. -1 is the correct error code there.

Reply:
I see. Also confirmed this by looking at the doc ;-)